flycheck.el

```diff
@@ -7735,22 +7749,20 @@ contains a cabal file."
     (`haskell-stack-ghc
      (or
       (when (buffer-file-name)
-        (locate-dominating-file (buffer-file-name) "stack.yaml"))
+        (flycheck--locate-dominating-file-matching
+         (file-name-directory
+          (buffer-file-name))
+         "stack.*\\.yaml\\'"))
```

Comment:
Bascially, the `\\'` means the same thing as `$` here. Namely, `$` is end of line and `\\'` is end of string.